src/LanguageServer/Impl/Diagnostics/DiagnosticsService.cs

```diff
@@ -90,16 +109,19 @@ internal sealed class DiagnosticsService : IDiagnosticsService, IDisposable {
         }
 
         private void PublishDiagnostics() {
+            KeyValuePair<Uri, IReadOnlyList<DiagnosticsEntry>>[] diagnostics;
             lock (_lock) {
-                foreach (var kvp in _diagnostics) {
-                    var parameters = new PublishDiagnosticsParams {
-                        uri = kvp.Key,
-                        diagnostics = kvp.Value.Select(ToDiagnostic).ToArray()
-                    };
-                    _clientApp.NotifyWithParameterObjectAsync("textDocument/publishDiagnostics", parameters).DoNotWait();
-                }
+                diagnostics = Diagnostics.ToArray();
```

Comment:
We aren't calling `_diagnostics.Clear();` anywhere. Is that correct?

Reply:
`ClearAllDiagnostics`, on Dispose. Otherwise it stays.